src/oph/ehoks/oppija/share_handler.clj

```diff
@@ -3,9 +3,78 @@
             [oph.ehoks.restful :as rest]
             [oph.ehoks.oppija.schema :as oppija-schema]
             [oph.ehoks.schema :as schema]
-            [oph.ehoks.db.db-operations.oppija :as db]
+            [oph.ehoks.db.db-operations.shared-modules :as db]
+            [oph.ehoks.db.postgresql.common :as cdb]
+            [oph.ehoks.db.postgresql.hankittavat :as hdb]
             [ring.util.http-response :as response]))
 
+(defn- get-tutkinnonosa-details [tyyppi uuid]
+  (cond
+    (= tyyppi "hato")
+    (db/select-hankittavat-ammat-tutkinnon-osat-by-module-id uuid)
+    (= tyyppi "hyto")
+    (db/select-hankittavat-yhteiset-tutkinnon-osat-by-module-id uuid)
+    (= tyyppi "hpto")
+    (db/select-hankittavat-paikalliset-tutkinnon-osat-by-module-id uuid)))
+
+(defn- combine-osaamisen-hankkiminen [uuid]
+  (let [module (db/select-osaamisen-hankkimistavat-by-module-id uuid)
+        tho (hdb/select-tyopaikalla-jarjestettava-koulutus-by-id
+              (:tyopaikalla-jarjestettava-koulutus-id module))]
+    (assoc
+      module
+      :muut-oppimisymparistot
+      (hdb/select-muut-oppimisymparistot-by-osaamisen-hankkimistapa-id
+        (:id module))
+      :tyopaikalla_jarjestettava_koulutus tho
+      :sovitut-tyotehtavat (hdb/select-tyotehtavat-by-tho-id (:id tho)))))
+
+(defn- combine-osaamisen-osoittaminen [uuid]
```

Comment:
Tähän osaamisen-osoittamisen ja osaamisen-hankkimistavan hakemiseen liittyy jonkin verran logiikkaa joten tämä ehkä kannattaa muuttaa käyttämään olemassaolevia funktioita esim oph.ehoks.hoks.hankittavat/get-hato-osaamisen-osoittaminen. Vastaava löytyy myös osaamisen hankkimistavalla mutta voi olla että noita ei ihan suoraan pysty käyttämään vaan pitää pikkusen taivuttaa. Ehkä antaa tämän olla nyt näin mutta muuttaa sitten kun lähdetään fronttia tekemään ja sieltä alkaa hahmottua tarpeet.

Reply:
Kaikki tein itse ja säästin -ratkaisut on siivottu pois ja nyt näissä käytetään pelkästään hankittavat-namespacen olemassaolevia toiminnallisuuksia; hakuihin piti tehdä erilliset funktiot module_id:llä hakemista varten.